torch/_dynamo/config.py

```diff
@@ -98,7 +98,7 @@
 
 # Record and write an execution record of the current frame to a file
 # if an exception is encountered
-replay_record_enabled = bool(os.environ.get("TORCH_COMPILE_DEBUG", False))
```

Comment:
This fails because `os.environ.get` returns a string and only empty string is `False`